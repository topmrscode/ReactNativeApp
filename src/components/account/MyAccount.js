
import React from 'react'
import {
    View,
    Text, 
    Button, 
    StyleSheet,
    TouchableOpacity
  } from 'react-native';
import BaseContainer from '../../assets/styles/Base'
import AsyncStorage from '@react-native-community/async-storage'

import {GetAccountAvatar} from '../../api/Account'
import {GetAccountSettings} from '../../api/Account'
import { Avatar } from '../../../node_modules/react-native-elements';

// -----------------------------------------
// GESTION ET AFFICHAGE DE LA PAGE PROFILE
// ----------------------------------------

class MyAccountScreen extends React.Component{
    constructor(props){
        super(props)
        this.state = {
            avatar : '',
            email : '',
            username : '',
        }
    }

    //  Appeller l'api imgur => recuperer l'avatar
    async componentDidMount(){
        
        // getSettings contient une promise qui ne sera finie qu'a la fin du then
        const getSettings = GetAccountSettings().then((data)=> {
            this.setState({email : data.data.email, username : data.data.account_url})
        })

        const getAvatar = GetAccountAvatar().then((data) => {
            this.setState({avatar : data.data.avatar})
        })

        // 'AWAIT' permet d'attendre que l operation declenchee par l'appel a
        // la fonction async 'ComponentDidMount 'soit terminee et donc de recuperer directement la valeur de retour de la promise (ici data)
        // ne peut etre utilisee que depuis une fonction asynchrome
        // getSettings contient une promise qui ne sera finie qu'a la fin du then

        // Les deux const sont des promesses
        // Pour attendre que les deux promesses aient finies de s'executer,
        // Il faut utiliser cette syntaxe.
        await Promise.all([getSettings, getAvatar])
    }
    // Deconnecter le user 
    _signOutAsync = async () => {
        await AsyncStorage.clear();
        this.props.navigation.navigate('Auth');
      };
    // Naviguer jusq'a ma page gallery 
    _imageGalleryAsync = async () => {
        this.props.navigation.navigate('MyGallery');
    };

    render() {
        return (
            <BaseContainer title={'Profil'} font={''} fontSize={20}>
                <View style={styles.container} >
                   <Avatar
                        size="xlarge"
                        rounded
                        activeOpacity={0.7}
                        source={{
                          uri:
                            this.state.avatar,
                        }}
                    />
                    <Text style={styles.text}>{this.state.username}</Text>
                    <Text>{this.state.email}</Text>
                    <TouchableOpacity style={styles.buttonConnect} onPress={this._imageGalleryAsync}>
                        <Text style={styles.buttonText}>My images</Text>
                    </TouchableOpacity>
                    <TouchableOpacity style={styles.buttonConnect} onPress={this._signOutAsync}>
                        <Text style={styles.buttonText}>Sign Out</Text>
                    </TouchableOpacity>
            
                </View>
            </BaseContainer>

        )
}

}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor:"#dfe3ee",
      fontFamily: 'Catalunyademo'
    },
    text: {
        marginTop: 20,
        fontFamily: 'Catalunyademo'
    },
    buttonConnect: {
        marginTop: 20,
        alignItems: 'center',
        borderRadius: 100,
        borderWidth: 1,
        borderColor: '#3b5998',
        backgroundColor: 'transparent',
        color: '#3b5998',
        paddingVertical: 10,
        paddingHorizontal: 15,
    },
      buttonText: {
        color: '#3b5998',
        textTransform: 'uppercase',
        fontSize: 15,
        fontWeight: 'bold',
        letterSpacing: 3,
    },
});


export default MyAccountScreen