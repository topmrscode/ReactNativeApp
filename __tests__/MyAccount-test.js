import React from 'react';
import MyAccount from '../src/components/Account/MyAccount.js';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';

// -------------------------------------------------------
// RECUPERER LES DONNES DEPUIS AXIOS / RENDU DES DONNEES  
// -------------------------------------------------------


// CONFIGURER LE TEST ------
jest.disableAutomock();

// mocker axios
jest.mock("axios");
// mocker le module dans le fichier correpondant 
jest.mock('../src/api/Account.js', () => ({
  // je donne la fonction du componentWillMount et ses parametres
  GetAccountAvatar: () => {
    // retourner ce que doit renvoyer la fonction
    return (
      new Promise((resolve, reject) => {
        var data = {
          data: {
            avatar: "http://www.monavatar.com/face.jpg"
          }
        }
        // Correspond au then qui est apres le GetAccountAvatar dans notre componentDidMount du componentt MyAccount
        resolve(data);
      })
  )},
  GetAccountSettings: () => {
    return (
      new Promise((resolve, reject) => {
        var data = {
          data: {
            email: "mon.email@domain.fr",
            account_url: "http://mon.domain.com"
          }
        }
        resolve(data);
      })
    )
  }
}));

Enzyme.configure({ adapter: new Adapter() });
test('renders correctly', async () => {
  wrapper = await (shallow(<MyAccount />));
  // Ma fonction va attendre que la ligne du dessus soit fini avant de continuer

  const tree = renderer.create(wrapper).toJSON();
  expect(tree).toMatchSnapshot();

  expect(wrapper.state('avatar')).toEqual('http://www.monavatar.com/face.jpg');
  expect(wrapper.state('email')).toEqual('mon.email@domain.fr');
  expect(wrapper.state('username')).toEqual('http://mon.domain.com');
});