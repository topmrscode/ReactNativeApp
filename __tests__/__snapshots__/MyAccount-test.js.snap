// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#323539",
          "justifyContent": "center",
          "padding": 10,
        },
        Object {
          "elevation": 3,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
          Object {
            "fontFamily": "",
            "fontSize": 20,
          },
        ]
      }
    >
      Profil
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dfe3ee",
        "flex": 1,
        "fontFamily": "Catalunyademo",
        "justifyContent": "center",
      }
    }
  >
    <View
      activeOpacity={0.7}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 75,
          "height": 150,
          "width": 150,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 75,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          onLoad={[Function]}
          source={
            Object {
              "uri": "http://www.monavatar.com/face.jpg",
            }
          }
          style={
            Object {
              "bottom": 0,
              "flex": 1,
              "height": null,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": null,
            }
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          transition={true}
        />
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "flex": 1,
                "height": null,
                "justifyContent": "center",
                "width": null,
              }
            }
            testID="RNE__Image__placeholder"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": null,
              "width": null,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "Catalunyademo",
          "marginTop": 20,
        }
      }
    >
      http://mon.domain.com
    </Text>
    <Text>
      mon.email@domain.fr
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#3b5998",
          "borderRadius": 100,
          "borderWidth": 1,
          "color": "#3b5998",
          "marginTop": 20,
          "opacity": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3b5998",
            "fontSize": 15,
            "fontWeight": "bold",
            "letterSpacing": 3,
            "textTransform": "uppercase",
          }
        }
      >
        My images
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#3b5998",
          "borderRadius": 100,
          "borderWidth": 1,
          "color": "#3b5998",
          "marginTop": 20,
          "opacity": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3b5998",
            "fontSize": 15,
            "fontWeight": "bold",
            "letterSpacing": 3,
            "textTransform": "uppercase",
          }
        }
      >
        Sign Out
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
